build.gradle

```diff
@@ -227,3 +231,34 @@ if (System.env.TRAVIS == 'true') {
         }
     }
 }
+
+def publishedProjects = subprojects.findAll()
+ 
+task jacocoRootReport(type: JacocoReport, group: 'Coverage reports') {
+    description = 'Generates an aggregate report from all subprojects'
+ 
+    dependsOn(publishedProjects.test)
+ 
+    additionalSourceDirs = files(publishedProjects.sourceSets.main.allSource.srcDirs)
```

Comment:
It is not clear to me what these lines are for.

Could you add some comments? e.g. reference online documentations?